lms/djangoapps/courseware/courses.py

```diff
@@ -470,3 +465,79 @@ def get_problems_in_section(section):
                     problem_descriptors[unicode(component.location)] = component
 
     return problem_descriptors
+
+
+def get_current_child(xmodule, min_depth=None, requested_child=None):
```

Comment:
@andy-armstrong This isn't a blocker on this PR, but I am curious about your thoughts on this.  This is more of a "taming" type question.